File: java/src/pattern/strategy/duck/DecoyDuck.java
package pattern.strategy.duck;

public class DecoyDuck extends Duck
{
    public DecoyDuck()
    {
	super.Duck();
	flyBehaviour = new FlyNoWay();
	quackBehaviour = new MuteQuack();
    }

    public void display()
    {
	System.out.println("I'm just a decoy duck!");
    }
}

File: php/src/pattern/strategy/duck/DecoyDuck.php
<?php

namespace pattern\strategy\duck;

class DecoyDuck extends Duck
{
    public function __construct()
    {
	parent::__construct();
	$this->_flyBehaviour = new FlyNoWay();
	$this->_quackBehaviour = new MuteQuack();
    }

    public function display()
    {
	echo "I'm just a decoy duck!\n";
    }
}
